app/api/auth/register/route.ts

```diff
@@ -5,6 +5,17 @@ import { NextResponse } from "next/server";
 export async function POST(request: Request) {
     try {
         const {email, password} = await request.json()
+        // Check if the email already exists in the database
+        const data: QueryResult = await sql`
+          SELECT email FROM users WHERE email = ${email}
+        `;
+        const userExists = data.rows.length > 0;
+        if (userExists) {
+          return NextResponse.json(
+            { error: "User with provided email already exists." },
```

Comment:
Responding with a message that indicates a user is already created with a given email exposes us to [user enumeration attacks](https://wiki.owasp.org/index.php/Testing_for_User_Enumeration_and_Guessable_User_Account_(OWASP-AT-002)).